### PR TITLE
feat(call-for-testing): allow per-snap testing docs

### PR DESCRIPTION
**Pull request by jnsgruk**
This change adds the ability to (slightly) customise the testing instructions - the start/end of the current testing instructions remain, as they are consistent across the estate, though I have updated the language slightly to make it more universal.

An example of an uncustomised issue: https://github.com/jnsgruk/ci-testing/issues/49 vs. a customised issue (see 3. under "how to test it manually): https://github.com/jnsgruk/ci-testing/issues/48 - and the change that triggered it: https://github.com/jnsgruk/ci-testing/commit/07b5a51c2c00b0dc2956f90b6e1c6f79a8950b49

### CODE REVIEW

**Comment by jnsgruk** [comment]
@soumyaDghosh I think this is what you were looking for? :)

**Comment by soumyaDghosh** [comment]
> @soumyaDghosh I think this is what you were looking for? :)

Yeah, this is it, but I was thinking of a different approach. I was thinking of a completely different `template.md` file because you can add new line here, but not completely remove some lines or is that too possible?

**Comment by soumyaDghosh** [comment]
> My thought on this change was that it'd enable you to add the _additional_ steps needed to test your SDK & content snap together

The testing steps for a content snap is not much different.

1. Stop the application using `ffmpeg-2204` (or we can list some apps like `tube-converter`, `tagger` etc)
2. Refresh (same as before)
3. Run the app and test some ffmpeg specific features (almost same as before)

But, for the sdk snap

1. Take a snap manifest that uses ffmpeg.
2. Change the version of `ffmpeg-2204-sdk` in the `build-snaps` part and rewrite it as `ffmpeg-2204-sdk/latest/candidate`
3. Build the snap and see if it succeeds.

**Comment by jnsgruk** [comment]
So I think that could work?

```yaml
jobs:
  call-for-testing:
    name: 📣 Create call for testing
    runs-on: ubuntu-latest
    steps:
      - name: 📣 Create call for testing
        uses: snapcrafters/ci/call-for-testing@main
        with:
          testing-instructions: |
            1. Modify a `snapcraft.yaml` which specifies `ffmpeg-2204-sdk` as a `build-snap`, and replace that with `ffmpeg-sdk-2204/latest/candidate`
            1. Rebuild the snap who's recipe you just updated 
```

**Comment by jnsgruk** [comment]
@soumyaDghosh I've changed it so the entire testing instructions section can be customised - I think this will be more useful to you? (If merging, this should be a squash merge)

File: call-for-testing/action.yaml [hunk 1]
```diff
@@ -34,6 +34,23 @@ inputs:
   store-token:
     description: "A token with permissions to upload to the specified channel"
     required: false
+  testing-instructions:
+    description: "Custom instructions for testing the `candidate` snap"
+    default: |
+      1. Stop the application if it was already running
+      1. Upgrade to this version by running
```
Comment (soumyaDghosh) [inline call-for-testing/action.yaml:41]:
```suggestion
      2. Upgrade to this version by running
```

File: call-for-testing/action.yaml [hunk 2]
```diff
@@ -34,6 +34,23 @@ inputs:
...
+
+        ```shell
+        snap refresh {{ env.snap_name }} --channel {{ env.channel }}
+        ```
+
+      1. Start the app and test it out.
```
Comment (soumyaDghosh) [inline call-for-testing/action.yaml:47]:
```suggestion
      3. Start the app and test it out.
```

File: call-for-testing/action.yaml [hunk 3]
```diff
@@ -34,6 +34,23 @@ inputs:
...
+      1. Finally, add a comment below explaining whether this app is working, and **include the output of the following command**.
```
Comment (soumyaDghosh) [inline call-for-testing/action.yaml:48]:
```suggestion
      4. Finally, add a comment below explaining whether this app is working, and **include the output of the following command**.
```

**Review by soumyaDghosh — CHANGES_REQUESTED** [review]

File: call-for-testing/action.yaml (hunk 1 above)
Comment (jnsgruk) [reply call-for-testing/action.yaml:41]: Ah, this was deliberate -- the Markdown processor on Github will automatically ensure the numbers are all in the right order :)

**Pull request merged**